mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -164,6 +169,13 @@ impl S3CrtClient {
         message.add_header(&Header::new("accept", "application/xml"))?;
         message.add_header(&Header::new("User-Agent", &self.user_agent_header))?;
 
+        match self.request_payer {
+            Some(ref payer) => {
+                message.add_header(&Header::new("x-amz-request-payer", payer))?;
+            },
+            None => {}
+        }
```

Comment:
We can make this a bit simpler as we don't care about the `None` case. [Reported by clippy check](https://github.com/awslabs/mountpoint-s3/actions/runs/4458017462/jobs/7846616113#step:6:83).

```suggestion
        if let Some(ref payer) = self.request_payer {
            message.add_header(&Header::new("x-amz-request-payer", payer))?;
        }
```